### PR TITLE
Added null propagation operator support

### PR DESCRIPTION
**Pull request by davidyaha**
To summary the discussion on #146.

### CODE REVIEW

File: experimental/null-propagation.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# [Null Propagation Operator](https://docs.google.com/presentation/d/11O_wIBBbZgE1bMVRJI8kGnmC6dWCBOwutbN9SWOK0fU/edit#slide=id.p)
```
Comment (adrianheine) [inline experimental/null-propagation.md]:
```suggestion
# [Null Propagation Operator](https://github.com/tc39/proposal-optional-chaining)
```

**Comment by adrianheine** [comment]
This does not yet address https://github.com/estree/estree/issues/146#issuecomment-365431078, right?

**Comment by JLHwung** [comment]
I think this PR should be closed in favor of #204

**Comment by nzakas** [comment]
Agreed. @rreverser?

**Comment by RReverser** [comment]
Yeah, makes sense. FWIW you don't need to wait for me for obvious cleanups / triage :)

**Pull request closed without merging**